x-pack/plugins/fleet/server/services/epm/packages/install.ts

```diff
@@ -534,9 +534,9 @@ export async function createInstallation(options: {
   const removable = !isUnremovablePackage(pkgName);
   const toSaveESIndexPatterns = generateESIndexPatterns(packageInfo.data_streams);
 
-  // For default packages, default the `keep_policies_up_to_date` setting to true. For all other
-  // package, default it to false.
-  const defaultKeepPoliciesUpToDate = DEFAULT_PACKAGES.some(
```

Comment:
The original intent here _should_ have been to leave `keep_policies_up_to_date` set to `undefined` unless it was explicitly set by the user, but this code introduced in https://github.com/elastic/kibana/pull/116993 sets the default value to `false` instead for any non-default package that's installed. 

Because we explicitly set this to `false` during installation for any non-default package, we can't reliably honor user settings during our migration. 

For example, if a user installed `apm` on their 7.16 Kibana instance, the installation SO would have `keep_policies_up_to_date: false`. So, we have to make a choice in this PR whether we should discard any user-provided value for `keep_policies_up_to_date` on managed policies and simply set them all globally to `true`.

Reply:
To close the loop here, I spoke with Josh online and got a few pieces of direction here. 

- The only packages that _must_ have their policies upgraded are `APM` and `Synthetics`
- We shouldn't automatically opt `system` into keeping policies up to date, because it's widely used and could cause rollout/performance issues in environments with many agents

With those two things in mind, I elected to make the changes detailed in my previous comment.